python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1642,6 +1642,38 @@ def wipe_asset(self, asset_key: AssetKey) -> None:
                 )
             )
 
+    def get_materialized_partitions(
+        self, asset_key: AssetKey, after_cursor: Optional[int] = None
+    ) -> Set[str]:
+        query = (
+            db_select(
+                [
+                    SqlEventLogStorageTable.c.partition,
+                    db.func.max(SqlEventLogStorageTable.c.id),
+                ]
+            )
+            .where(
+                db.and_(
+                    SqlEventLogStorageTable.c.asset_key == asset_key.to_string(),
+                    SqlEventLogStorageTable.c.partition != None,  # noqa: E711
+                    SqlEventLogStorageTable.c.dagster_event_type
+                    == DagsterEventType.ASSET_MATERIALIZATION.value,
+                )
```

Comment:
Why not put the after_cursor filter in here? Seems like that might save a bit of trouble for the database, but perhaps the query planner unwinds all of that anyway.

Reply:
Yeah, I'm pretty sure it doesn't matter... SqlAlchemy constructs the query correctly anyway.

It can't go in this current block because we only want to apply it if a `after_cursor` arg is supplied.

Reply:
hm yeah, was thinking of it like it was a subquery but you're right it shouldn't matter